### DISCUSSION TITLE
Google Storage gRPC public API

### DISCUSSION THREAD

**Discussion by lromor** [googleapis/googleapis#772] [General]
Hi!

I wanted to play a bit with the gRPC API for google cloud storage, but I got a:
```
Requested bucket, 'foo.bar', is not allowed to access the GCS gRPC API. Note: this API is currently in testing, and is not yet available for general use.
```

I'm just curious if there's a public ETA for this? 

-l

**Comment by coryan** [comment]
None that I am allowed to share here, sorry.  If you have a technical account manager or are in contact with another support person they may be able to give you an answer or put you in touch with the product managers.

**Reply by wspeirs** [reply]
I'm running into the same issue... any idea _when_ it'll be made public?

**Reply by psupriya1991** [reply]
I am facing the issue. Any chance, you found the alternative?

**Reply by zk-gt** [reply]
Hi, I am a PM on the Google Cloud Storage team. At present, Google Cloud Storage gRPC API is Generally Available (GA) with Allowlist. This means that users must have their GCS buckets allowlisted before they are able to utilize the API. Please contact gcs-grpc-contact@google.com with a list of the buckets that you would like to be Allowlisted

**Reply by jeff-brooks** [reply]
Hey @zk-gt, thanks for providing that info!

Is there an announcement about the api becoming GA anywhere? Or documentation on the allowlist process?

**Reply by zk-gt** [reply]
The launch is targeted for later this year and there will be public blogpost to announce the launch. 
For allowlist, you have to send an email with a list of buckets to [gcs-grpc-contact@google.com](mailto:gcs-grpc-contact@google.com).

**Comment by KipHamiltons** [comment]
I emailed the email above and it's hit public GA.
Here's a response I got (along with a nice one from one of the engineers 😄)

> We're very excited to announce the Public GA of the Cloud Storage gRPC API and removal of Allowlist. This high-performance API provides a new way to interact with Cloud Storage and has been refined through an allowlist GA with customers like you.
>
> What is gRPC?
> [gRPC](https://grpc.io/), a high-performance, open-source universal RPC framework developed by Google, that employs Protocol Buffers for service definition. When implemented with Cloud Storage, gRPC leverages [direct connectivity](https://cloud.google.com/vpc-service-controls/docs/set-up-private-connectivity#direct-connectivity) between Compute Engine and Cloud Storage buckets. This configuration bypasses [Google Front Ends](https://cloud.google.com/docs/security/infrastructure/design#google-frontend-service)  (GFEs), resulting in optimized network routing, lower resource requirements, and improved performance. Customers can access gRPC through the: 
[Cloud Storage connector for Dataproc ](https://cloud.google.com/dataproc/docs/concepts/connectors/cloud-storage#connect-to-cloud-storage-using-grpc)
[C++/Java/Go Client Libraries](https://cloud.google.com/storage/docs/enable-grpc-api), or the 
[Apache Beam connector for Dataflow](https://cloud.google.com/dataflow/docs/guides/read-from-cloud-storage#enable-connector-grpc)
>
> Key Benefits:
> Improved read latencies : gRPC can significantly improve the performance of analytics workloads that involve large amounts of data, by providing faster read latencies.
>
> Customers can provide feedback or ask questions through account teams or directly via [gcs-grpc-contact@google.com](mailto:gcs-grpc-contact@google.com).
